procsimulator.py

```diff
@@ -1361,6 +1398,43 @@ def execInstr(self):
                 for flag in workingFlags:
                     self.flags[flag] = workingFlags[flag]
 
+        elif t == InstrType.multiplylong:
+            op1 = self.regs[misc['operandsmul'][0]].get()
+            op2 = self.regs[misc['operandsmul'][1]].get()
+            destHi = misc['rdHi']
+            destLo = misc['rdLo']
+
+            res = (self.regs[destHi].get() << 32) + self.regs[destLo].get()
+            if misc['signed']:
+                if op1 & 1 << 31:
```

Comment:
Technically, these conditions could be replaced by something like:
`op1 -= (op1 >> 31 & 1) << 32`
Not sure if it's readily understandable though...